### PR TITLE
Update docs to add latest explain and give examples of latest changes

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
Docs updates

### Which issue(s) this PR fixes

N/A

### Describe testing done for PR

N/A

### CODE REVIEW

File: docs/commands-details/workload_create_update_apply.md [hunk 1]
```diff
@@ -402,17 +407,17 @@ tanzu apps workload apply tanzu-java-web-app --git-repo https://github.com/vmwar
...
 
-Commit in Git repository from where the workload is resolved. Can be used with `--git-branch` or `git-tag`.
+Commit in Git repository from where the workload is resolved. Either `--git-branch` or `--git-tag` can be specified with it too.
+It can be unset by defining it as empty string when applying a workload (`--git-commit ""`).
 
 <details><summary>Example</summary>
 
 ```bash
-tanzu apps workload apply tanzu-java-web-app --git-repo https://github.com/vmware-tanzu/application-accelerator-samples --sub-path tanzu-java-web-app --git-tag tap-1.3 --git-commit 1c4cf82e499f7e46da182922d4097908d4817320 --type web
+tanzu apps workload apply tanzu-java-web-app --git-repo https://github.com/vmware-tanzu/application-accelerator-samples --sub-path tanzu-java-web-app --git-commit 1c4cf82e499f7e46da182922d4097908d4817320 --type web
```
Comment (shaheerkootteeri) [inline docs/commands-details/workload_create_update_apply.md:420]: --git-tag  needed here for this example?

File: docs/commands-details/workload_create_update_apply.md [hunk 2]
```diff
@@ -798,12 +802,79 @@ tanzu apps workload apply tanzu-java-web-app --git-repo https://github.com/vmwar
...
+    ref:
+      apiVersion: rabbitmq.com/v1beta1
+      kind: RabbitmqCluster
+      name: example-rabbitmq-cluster-1
+  source:
+    git:
+      ref:
+        branch: main
+      url: https://github.com/jhvhs/rabbitmq-sample
+status:
```
Comment (shaheerkootteeri) [inline docs/commands-details/workload_create_update_apply.md:872]: Do we need this status section ?

File: docs/commands-details/workload_create_update_apply.md (hunk 1 above)
Comment (warango4) [reply docs/commands-details/workload_create_update_apply.md:420]: No, the idea is to let users know they do not need to provide tag or branch when using a commit. It was a bug that we had previously.

File: docs/commands-details/workload_create_update_apply.md (hunk 2 above)
Comment (warango4) [reply docs/commands-details/workload_create_update_apply.md:872]: that's an actual output, is shown like that, I just copy-pasted it.

**Review by shaheerkootteeri — APPROVED** [review]
LGTM

File: docs/commands-details/workload_create_update_apply.md (hunk 2 above)
Comment (shaheerkootteeri) [reply docs/commands-details/workload_create_update_apply.md:872]: I reckon we want to display the exact output of the command, so marking as resolved

**Review by odinnordico — APPROVED** [review]

**Review by anibmurthy — APPROVED** [review]

**Pull request merged**